### ISSUE TITLE
does not work with koa-router@7.0.1

### ISSUE THREAD

**Issue by chovy** [kudos/koa-websocket#14]
koa-router@7.0.1 is the new koa2 router.

The example for koa-router only works with 5.x. I've created an example. If you switch between 7.x and 5.x you can see it will stop working on 7.x

```
//index.js
require("babel-register");
require("./app.js");
```

```
//.babelrc
{
  "presets": ["es2015-node5"],
  "plugins": [
    "transform-async-to-generator",
    "syntax-async-functions"
  ]
}
```

```
//app.js
var koa = require('koa');
var websockify = require('koa-websocket');
var router = require('koa-router');
var app = new koa();

var api = router();
var socket = websockify(app);

api.get('/*', function* (next) {
  const ctx = this;
  ctx.websocket.send('Hello World');
  ctx.websocket.on('message', function(message) {
    console.log(message);
  });
});

app.ws.use(api.routes()).use(api.allowedMethods());
app.listen(3000);
```

Start with `node ./index.js`

**Comment by kudos** [comment]
@chovy I'm still using Koa 1.x in my projects for now, but I'll happily accept patches for koa-router@7.0.1 and would maintain a koa-websocket@2.0.0 from there.

**Comment by chovy** [comment]
@kudos -- thanks. I'm happy to submit a patch if I knew what the fix is.

I believe its this line here, but I'm not sure how to make this async in koa2.

https://github.com/kudos/koa-websocket/blob/master/index.js#L27

**Comment by anthonyettinger** [comment]
@kudos any idea?

**Comment by donjae** [comment]
@anthonyettinger Im having the same issue too, and the workaround at this point seems really painful (using old router/breaking some of the IoC in my application). I tried patching it myself by updating some of the dependencies, but didn't have any luck.

**Comment by cymen** [comment]
I opened a PR for this here:

https://github.com/kudos/koa-websocket/pull/18

The biggest help was using `DEBUG=koa* node test.js` as that emits more information from Koa. That additional logging made it easier to track down what wasn't working.

**Comment by kudos** [comment]
Thanks to @cymen this is now resolved! Update this to 3.0.1 if you're running a Koa 2 stack.
